hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/replication/VerifyReplication.java

```diff
@@ -788,6 +946,29 @@ private static void printUsage(final String errorMsg) {
         + "2181:/cluster-b \\\n" + "     TestTable");
   }
 
+  private static ExecutorService buildReCompareExecutor(int maxThreads, Mapper.Context context) {
+    return new ThreadPoolExecutor(
+      0,
+      maxThreads,
+      1L,
+      TimeUnit.SECONDS,
+      new SynchronousQueue<>(),
+      buildRejectedReComparePolicy(context)
+    );
+  }
+
+  private static CallerRunsPolicy buildRejectedReComparePolicy(Mapper.Context context) {
+    return new CallerRunsPolicy() {
+      @Override
+      public void rejectedExecution(Runnable runnable, ThreadPoolExecutor e) {
+        LOG.info("Re-comparison execution rejected. Running in main thread.");
```

Comment:
might want to make this debug. In general best to avoid info logs in M/R jobs because they can unexpected slow down jobs if they start spamming